### ISSUE TITLE
git lfs fsck cannot find the file specified

### ISSUE THREAD

**Issue by helt** [git-lfs/git-lfs#3511]
`git lfs fsck` fails just like every other command `git lfs` and also `git push`. I have absolutely no idea where to get the file in question from, nor how to fix this issue.

Now, extraordinary stuff that happend on the repo recently is (a) I pulled in an updated .gitattributes and (b) i ran `git gc`, which iirc finished with some error i accidentially ignored. 

```
$ git lfs logs last
git-lfs/2.6.1 (GitHub; windows amd64; go 1.11.1; git dc072c3e)
git version 2.20.1.windows.1

$ git-lfs fetch --all
Could not scan for Git LFS files

open C:\Users\somewhere\.git\objects\pack\pack-19d35aff7d741a3bcd25c9d759959e1676fe7c88.idx: The system cannot find the file specified.

Current time in UTC:
2019-02-05 22:16:25

ENV:
LocalWorkingDir=C:\Users\somewhere
LocalGitDir=C:\Users\somewhere\.git
LocalGitStorageDir=C:\Users\somewhere\.git
LocalMediaDir=C:\Users\somewhere\.git\lfs\objects
LocalReferenceDirs=
TempDir=C:\Users\somewhere\.git\lfs\tmp
ConcurrentTransfers=3
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=C:\Users\somewhere\.git\lfs
AccessDownload=basic
AccessUpload=basic
DownloadTransfers=basic
UploadTransfers=basic
GIT_EXEC_PATH=C:/Program Files/Git/mingw64/libexec/git-core

```

For completeness, also the git push error:
```
$ git push
Locking support detected on remote "origin". Consider enabling it with:
  $ git config lfs.https://git.exmaple.com/foo/bar.git/info/lfs.locksverify true
ref feature/foo:: open C:\foobar\.git\objects\pack\pack-19d35aff7d741a3bcd25c9d759959e1676fe7c88.idx: The system cannot find the file specified.
error: failed to push some refs to 'https://git.exmaple.com/foo/bar.git'
```
Any idea where to go from here?

**Comment by bk2204** [comment]
Hey, sorry to hear you're having trouble.

It looks like you have a pack but no index file, which is going to prevent Git LFS from working. This should clear up if you run `git index-pack C:\Users\somewhere\.git\objects\pack\pack-19d35aff7d741a3bcd25c9d759959e1676fe7c88.pack`. That will rebuild the index file and Git LFS will then be able to read the repository as normal.

**Comment by helt** [comment]
Thanks bk, your fix worked perfectly. 

So I was right when guessing that git lfs is not working as intended here.

**Comment by bk2204** [comment]
I don't think it's that Git LFS is not working as intended here, I think it's that your repository is missing index files from a bad `git gc`. Generally, both Git and Git LFS require you to be able to read the index files in order to read the packs unless you're actually indexing the packs with `git index-pack` or something that calls it (like `git gc`).

It's definitely the case that Git LFS won't work in this case, but a repository that's missing the index files is generally considered broken. Much of Git probably won't work, either.

**Comment by salda** [comment]
I remember our team somehow started using Git LFS and I have similar, but worse problem. 
I did nothing with the Git LFS, I just merged develop into my branch and fetching, pulling and committing worked, but now, I needed to push so:
$ git push
ref <branch_name>:: open <repository_path>\git\.git\objects\pack\pack-000b0e9b4b0b70f213c8446df15028c04a7f6486.idx: The system cannot find the file specified.
error: failed to push some refs to '<git_address>'. 
So I tried your quickfix:
$ git index-pack <repository_path>\git\.git\objects\pack\pack-000b0e9b4b0b70f213c8446df15028c04a7f6486.pack
fatal: cannot open packfile '<messed_path>-000b0e9b4b0b70f213c8446df15028c04a7f6486.pack': No such file or directory

Hm, now I see, it probably misses double quotation marks around the path...

After that I have another idx problem, but I guess with time, I fix them all.
